### DISCUSSION TITLE
What's left before scala-cli can be introduced on the official Scala Getting Started page?

### DISCUSSION THREAD

**Discussion by bjornregnell** [VirtusLab/scala-cli#905] [Ideas]
After discussions in the new forum for teachers [here](https://teachers.scala-lang.org/t/simpler-local-scala-install-and-execution/27/4), I'm opening this discussion to pave the way for introducing scala-cli on the official getting started page, by collecting links to relevant discussions and issues and also discuss roadmap and current potential show-stoppers etc. 

@julienrf suggests to make the migration of scala-cli to the [official getting-started-page](https://docs.scala-lang.org/getting-started/index.html) in a *step-wise manner*:
1. Make scala-cli on feature parity with coursier, esp. by enabling installation abilities
2. Replace the official `scala` runner by `scala-cli`
3. Update the instructions on the getting-started page to enable one-click install of the new `scala` for all platforms
4. ... ?

Related issues and discussions:
* https://github.com/VirtusLab/scala-cli/issues/904
* https://github.com/VirtusLab/scala-cli/discussions/351
* https://github.com/VirtusLab/scala-cli/issues/349
* https://github.com/VirtusLab/scala-cli/discussions/593
* https://teachers.scala-lang.org/t/simpler-local-scala-install-and-execution/27/5
* https://github.com/VirtusLab/scala-cli/discussions/279

**Comment by bjornregnell** [comment]
@romanowski @anatoliykmetyuk @alexarchambault  @prolativ After discussions in the new teacher forum, @julienrf asked me to initiate a discussion between Scala Center and VirtusLab on this. I chose to make it a discussion topic here, if you agree that this is a good starting point for collecting input? 

I'm also happy to arrange a Zoom meeting or similar to discuss things live if you think that can be useful. My interest in this is to get the new getting-started experience ready before the summer update of teaching material before the fall semester starts. I guess more teachers have the fall semester as a target for introducing scala-cli to students...

**Comment by bjornregnell** [comment]
@julienrf I outlined your "step-wise" approach to migrating the official getting-started page from cs to scala-cli above. Did I capture your ideas on this as intended?

**Comment by bjornregnell** [comment]
Another related issue to become in feature parity with the old scala runner: https://github.com/VirtusLab/scala-cli/issues/466

**Comment by SethTisue** [comment]
> Another related issue to become in feature parity with the old scala runner: https://github.com/VirtusLab/scala-cli/issues/466
 
and #753

**Comment by bjornregnell** [comment]
I think the getting-started page is really difficult to get right as we need to cater for so many different potential visitors, e.g.:
* Professionals that don't know Scala
* Professionals that already know Scala 2 and sbt and wants to try out Scala 3
* University students learning how to program
* University students taking an advance course in computer science
* University teachers using Scala as a tool for teaching X
* Kids in school
* Teachers of kids in school teaching computational thinking

I'm fine with not prioritizing kids over the others, but I do think that we should prioritize beginners to Scala, esp. students, as the Scala adoption in the long run is dependent on students (I think that's one reason that Java took off in the late 90'ies: for a long period a lot of universities used Java as a first language when teaching computer science and software engineering; the risk is that Python will take over that but I hope that Scala eventually can play a significant role as a teaching tool on planet Earth and beyond :)).

The current official Scala getting-started page tries to encompass several types of users and use cases, and that's fine but I think the top of the page should be a simple button for each platform with a one click install and then an instruction on how to run a hello world program and links to *How to open a terminal in* [Windows](https://docs.microsoft.com/en-us/windows/terminal/install), [MacOS](https://support.apple.com/guide/terminal/open-or-quit-terminal-apd5265185d-f365-44cb-8b09-71a064a42125/mac), and [Linux Unbuntu](https://ubuntu.com/tutorials/command-line-for-beginners#3-opening-a-terminal)

For Scala-beginners, I do think it is vital not to confuse them with too many unknown concepts (sbt, coursier, Scastie. Metals) and on the top of the page offer a simple button for each platform with a guess of a marked default, similar to [scala-cli's install page](https://scala-cli.virtuslab.org/install). 

Then we can have use-case-based texts or links if you e.g. want a build tool or try out code directly in the browser, use a specific editor, etc.

**Reply by bishabosha** [reply]
thank you for this advice, this will be very useful for updating the current getting-started page!

**Comment by bjornregnell** [comment]
Some more input in this comment here @bishabosha :
https://github.com/VirtusLab/scala-cli/issues/904#issuecomment-1114687827
